go/vt/vtgate/vstream_manager.go

```diff
@@ -179,13 +179,8 @@ func (vsm *vstreamManager) resolveParams(ctx context.Context, tabletType topodat
 		return nil, nil, nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "vgtid must have at least one value with a starting position")
 	}
 	// To fetch from all keyspaces, the input must contain a single ShardGtid
-	// that has an empty keyspace, and the Gtid must be "current". In the
-	// future, we'll allow the Gtid to be empty which will also support
-	// copying of existing data.
+	// that has an empty keyspace.
 	if len(vgtid.ShardGtids) == 1 && vgtid.ShardGtids[0].Keyspace == "" {
-		if vgtid.ShardGtids[0].Gtid != "current" {
-			return nil, nil, nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "for an empty keyspace, the Gtid value must be 'current': %v", vgtid)
-		}
 		keyspaces, err := vsm.toposerv.GetSrvKeyspaceNames(ctx, vsm.cell, false)
 		if err != nil {
 			return nil, nil, nil, err
```

Comment:
I would prefer the request be more explicit as it will likely generate A LOT of load and could have a big impact on production. I think it would also be nice if the keyspace wildcard selection lined up with the table wildcard selection in `binlogdata.Filter.Rule.Match`, which is `/.*` as a valid regular expression is accepted if the string has a `/` prefix. So I think accepting valid regular expressions for the keyspace would be much nicer: `/.*`.

Reply:
@mattlord It makes sense in terms of the difference in impact between "streaming" and "copy."
Accepting the keyspace wildcard sounds useful. With this feature, we don't have to accept the request with an empty keyspace and empty gtid.

Do you think the following specifications are still required?

- A special request flag, for example, "EnableEmptyKeyspace" or "EnableWildcardKeyspace" in VStreamFlags.
   - I think only the keyspace wildcard like `/.*` is enough.
- The valid combination of an empty keyspace and "current" gtid.
   - While the combination of an empty keyspace and the gtid other than "current" is invalid. It complicates the code, but not so much.
   - Some may think it's better to stop accepting this input for the sake of simplicity at this moment.

As such, I plan to update accepting the keyspace wildcard and preserve the previous behavior as much as possible like the below table. What do you think?

| gtid  | keyspace | valid or invalid(=noop) |
|-------|-------|--------------|
| "" | ""    | invalid because of no literal match. No change.     |
| "current" | ""       | valid because of the compatibility. No change.      |
| "" | "/.*"       | valid if found because of wildcard match. New behavior.      |
| "current" | "/.*"       | valid if found because of wildcard match. New behavior.      |
| "" | "/ks.*/"       | valid if found because of wildcard match. New behavior.      |
| "current" | "/ks.*/"       | valid if found because of wildcard match. New behavior.      |

Reply:
I think the explicit keyspace wildcard is enough and we don't need a new flag around that.

Now that I understand the usage of `current` better, I think we have to keep it in place for compatibility sake. What do you think @rohit-nayak-ps ?

Reply:
Hi @rohit-nayak-ps, I'd appreciate your feedback. Let me know if anything is missing.

Reply:
@yoheimuta, sorry for the delay. I am fine with the approach outlined above.

We should add a (or modify the existing) e2e test to confirm that resharding will continue to work if multiple keyspaces are being streamed.  We have code (see the logic around `journaler` in `vstream_manager`), which waits for participating shards to converge, before automatically continuing to stream from the new shards.  I think it should be fine, when there are multiple keyspaces, but confirming with a test will be useful.

Reply:
Thank you for your confirmation!
I'm looking into e2e test too.

Reply:
Hi 👋 

@mattlord I implemented the keyspace wildcard selection by [d83daaf](https://github.com/vitessio/vitess/pull/11909/commits/d83daaf0a244aac6c69d34d709c5fe76e9cdc640) and [c39af7c](https://github.com/vitessio/vitess/pull/11909/commits/c39af7ce7379d93b5330ad78919cfc0b908e7d5a).

@rohit-nayak-ps I added the e2e test by [71e1681](https://github.com/vitessio/vitess/pull/11909/commits/71e16813dc87d980e80a51636874124e5338a8f2) and [436568d](https://github.com/vitessio/vitess/pull/11909/commits/436568de9f2fe7d5aedabdf86dea6108ce3e2243)

And I think the CI failure is caused by some unreliable tests because the failing test changes every time we run it and it passes when we run it locally.

Reply:
Thanks @yoheimuta, we will review it soon. Yes, that CI failure is also happening on other PRs and I have asked the related team to look at it.